internal/backends/sqlite/database.go

```diff
@@ -95,6 +95,26 @@ func (db *database) DropCollection(ctx context.Context, params *backends.DropCol
 	return nil
 }
 
+// CreateCollection implements backends.Database interface.
+func (db *database) Ping(ctx context.Context) error {
+	sqlDB := db.r.DatabaseGetExisting(ctx, db.name)
+
+	// create temporary non-existent database to make sure that it's possible for user to create one
+	// (there might be some fs permission errors)
+	if sqlDB == nil {
+		var err error
+
+		sqlDB, err = db.r.DatabaseGetOrCreate(ctx, db.name)
+		if err != nil {
+			return lazyerrors.Error(err)
+		}
+
+		defer db.r.DatabaseDrop(ctx, db.name)
+	}
+
+	return sqlDB.PingContext(ctx)
+}
+
```

Comment:
I decided to create `Ping` as a part of database interface to use native backends ping commands in place of "implementing" pings in our way, feel free to discuss this approach.

Reply:
I don't think we should do that. On the one hand, our interface is "fat" enough to allow efficient implementations – for example, `Query` is very flexible in what it pushes down and not. On the other hand, it should be "thin" enough to avoid extra implementation. For example, `ping` command handle could just call `ListDatabases` – that would check that database exists but does not require each backend to implement one extra command.

Reply:
We cannot guarantee if `ListDatabases` implementation for specific backend does actually initialize any connection. Even in SQLite `ListDatabases`, doesn't query anything from sqlite module, but access existing `Pool` struct with stored `dbs` in it.

Reply:
I wonder what do we expect from ping? My understanding for now is that we want to ensure that it's possible to have any connectivity with the backend. With Postgres it's easy as this connectivity is just connection over socket/network, but SQLite? What do we want to ensure?